tests/results/DiffTest.py

```diff
@@ -161,6 +166,14 @@ def test_addition_rename(self):
         uut.rename = "other.py"
         self.assertRaises(ConflictError, other.__add__, uut)
 
+    def test_addition_filename(self):
+        uut = Diff(self.file, filename="file")
+        other = Diff(self.file, filename="file")
+        self.assertEqual((other + uut).filename, "file")
+
+        other.filename = "elif"
+        self.assertRaises(ConflictError, other.__add__, uut)
+
     def test_from_string_arrays(self):
         a = ["q", "a", "b", "x", "c", "d"]
         b = ["a", "b", "y", "c", "d", "f"]
```

Comment:
By everything related, do you mean split this into two commits, one for Result and ResultTest, and the other for Diff and DiffTest?